bugbot/crash/analyzer.py

```diff
@@ -540,17 +555,32 @@ def fetch_representative_processed_crash(self) -> dict:
             self._fetch_crash_reports(self.top_proto_signature, self.top_build_id),
             self._fetch_crash_reports(self.top_proto_signature, self._build_ids()),
         )
-        for report in reports:
+
+        first_representative_report = None
+        for i, report in enumerate(candidate_reports):
             uuid = report["uuid"]
             processed_crash = socorro.ProcessedCrash.get_processed(uuid)[uuid]
             if (
-                not limit_to_top_proto_signature
-                or processed_crash["proto_signature"] == self.top_proto_signature
+                limit_to_top_proto_signature
+                and processed_crash["proto_signature"] != self.top_proto_signature
             ):
-                # TODO(investigate): maybe we should check if the stack is
-                # corrupted (ask gsvelto or willkg about how to detect that)
+                continue
+
+            if first_representative_report is None:
+                first_representative_report = processed_crash
+
+            if not self._is_corrupted_crash_stack(processed_crash):
                 return processed_crash
 
+            if i >= 10:
```

Comment:
Based on personal experience, I'd have gone for 15 or 20. Corrupted stacks can occur when memory is constrained and the chances are rather slim to get a stack which is not corrupted. Corrupted stack rates of 80% are not uncommon.

Reply:
Thank you! Done in ec1d7db8da96e41501fedf4ea91ce57cb73ec414.